### PR TITLE
Converted to python3 and solved some bugs

### PR DESCRIPTION
**Pull request by JohnElmasry**
I converted the code into a python3 compatible code. Also a bug has been solved when a cell breaks more than one validation rule. For example, Cell H4 in my proposed excel file has an error of conditional and datatype. The old code used to crash over this cell. In this pull request this bug has been handled.

### CODE REVIEW

File: excel_validator.py [hunk 1]
```diff
@@ -178,9 +182,13 @@ def validate(settings, excelFile, sheetName, tmpDir, printErrors = False):
                 errors.append((coordinates, ValueError))
 
             #find header (first) row
+            #if header exits " ID " awl cell fe elheader
             if settings['header'] != True:
+                # if value bysawy awl cell fe el header
```
Comment (mkesicki) [inline excel_validator.py]: ??

Comment (mkesicki) [inline excel_validator.py]: ??

File: excel_validator.py [hunk 2]
```diff
@@ -178,9 +182,13 @@ def validate(settings, excelFile, sheetName, tmpDir, printErrors = False):
                 errors.append((coordinates, ValueError))
 
             #find header (first) row
+            #if header exits " ID " awl cell fe elheader
             if settings['header'] != True:
+                # if value bysawy awl cell fe el header 
                 if value == settings['header']:
+                    #Saffar el Header
```
Comment (mkesicki) [inline excel_validator.py]: ??

File: excel_validator.py [hunk 3]
```diff
@@ -236,13 +247,19 @@ def isEmpty(row):
...
     try:
         results = validate(settings, args.file, args.sheetName, args.tmpDir, args.errors)
-    except Exception, e:
-        sys.exit("Error occured: " + e.message)
+    except Exception as e:
+        sys.exit("Error occured: " + str(e))
 
+#result = true >> elfile saleem w validated
+#result != True and not equal None "Hatly esm el result file"
+#results !=True and equal None " File Kbeer "
```
Comment (mkesicki) [inline excel_validator.py]: these 3 lines can be removed

File: excel_validator.py [hunk 4]
```diff
@@ -236,13 +247,19 @@ def isEmpty(row):
...
     if results != True:
         if results:
             sys.exit("Validation errors store in: [[" + results + "]]")
         else:
             sys.exit("Invalid file is too big to generate annotated Excel file")
 
     sys.exit(0)
+
+
+
```
Comment (mkesicki) [inline excel_validator.py]: these lines can be removed

File: validate.bat [hunk 5]
```diff
@@ -1,4 +1,4 @@
 echo %1
-python excel_validator.py "example/example.yml" %1 "Example" "C:\temp" --errors=true
+python excel_validator.py  "example/example.yml" C:\Users\john.elmasry\excel_validator\example\excel.xlsx "Example" "." --errors=true
+pause
```
Comment (mkesicki) [inline validate.bat:3]: now will be hardcoded value to file name, btw orginal version with temp value also looks fishy :)

**Review by mkesicki — CHANGES_REQUESTED** [review]

**Comment by mkesicki** [comment]
@JohnElmasry , thank you for PR. I think readme file should be updated also ;)

**Comment by JohnElmasry** [comment]
Hello, Thank you for reaching out, and thanks for the program

Kindly find edited code

**Pull request merged**